Sources/MapboxNavigation/NavigationMapView+BuildingHighlighting.swift

```diff
@@ -11,17 +11,93 @@ extension NavigationMapView {
      
      - parameter coordinates: Coordinates which represent building locations.
      - parameter extrudesBuildings: Switch which allows to highlight buildings in either 2D or 3D. Defaults to true.
-     
-     - returns: Bool indicating if number of buildings found equals number of coordinates supplied.
+     - parameter completion: A closure that operates after the buildings are highted.
+
      */
-    @discardableResult public func highlightBuildings(at coordinates: [CLLocationCoordinate2D], in3D extrudesBuildings: Bool = true) -> Bool {
-        return false
+    public func highlightBuildings(at coordinates: [CLLocationCoordinate2D],
+                                   in3D extrudesBuildings: Bool = true,
+                                   completion: () -> Void) {
+        let group = DispatchGroup()
+        guard let identifiers = try? mapView.__map.getStyleLayers().compactMap({$0.id}).filter({ $0.contains("building") }) else { return }
+
+        let _ = coordinates.map({
+            group.enter()
+            let screenCoordinate = mapView.point(for: $0, in: self)
+            mapView.visibleFeatures(at: screenCoordinate,
+                                    styleLayers: Set(identifiers),
+                                    completion: { [weak self] result in
+                                        guard let validSelf = self else { return }
+                                        if case .success(let features) = result {
+                                            if let identifier = features.first?.featureIdentifierToDouble() {
+                                                self?.foundBuildingIds.insert(identifier)
+                                            }
+                                            group.leave()
+                                        }})
+        })
+
+        group.notify(queue: DispatchQueue.main) {
```

Comment:
You should wait till all operations complete, instead of notifying right away. After finishing all operations completion block can be called with boolean parameter which will tell whether all identifiers were found.

Reply:
Thanks for the suggestion, I'll update this part.